canonical/maas-ui
# fix(UI): Node summary page

## Note

This is a combination of two PR's which have already been code reviewed: #4350 and #4366. This PR is for QA purposes since they each affect design on the same page.

## Done

- Extend vertical grid dividers to top of summary card
- Add padding on either side of horizontal divider on summary card
- Details items (Owner, Host, Zone etc) have a max gap of 1rem
- If there is extra space, tags are allowed to stop truncating first
- Align test failed text to other links in top half of overview
- Redistribute space after tags finish truncating
- Hide "Host" on physical machines that don't have a VM host
- Added "BIOS boot mode" to the node hardware information card
- Removed "Domain" from the node overview card, since it's displayed in the header next to the node name
- Migrated DetailsCard.test to RTL 
- Migrated HardwareCard.test to RTL and removed need for snapshots
- Migrated NumaCard.test to RTL and removed need for snapshots
- Migrated NumaCardDetails.test to RTL and removed need for snapshots
- Added lines in between list items on the Numa Node and Hardware Information cards
- Removed lines between card titles and content
- Shrunk top padding on all cards in Machine Summary
- Left-aligned list item labels in Numa Node and Hardware Information cards

## QA

### PR demo
You can see a live deployment of this branch here: https://maas-ui-4377.demos.haus/

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Navigate to Machines and click a machine
- Stretch the window horizontally and ensure that the tags are allowed to un-truncate
- Squeeze the window horizontally and ensure none of the details (labels or info) are wrapping, and that the tags are truncating
- Ensure that the boot mode displays in the hardware info card. _Note: this may display as `Unknown` if no mode is set_
- Ensure that dividers are displayed between items in the hardware info and numa node cards
- Navigate back to the machines list and find a machine without a VM host
- Ensure that there is no "host" field shown in the overview details

## Fixes

-Fixes https://github.com/canonical/app-tribe/issues/1281 
-Fixes #1595

## Launchpad issue

`lp#1893670`
https://bugs.launchpad.net/maas/+bug/1893670

## Screenshots

### Before
![Screenshot from 2022-09-01 10-12-48](https://user-images.githubusercontent.com/35104482/187877996-1a7eeb2e-be36-4b99-bf26-4d22a5e013c3.png)

### After
![Screenshot from 2022-09-01 10-12-58](https://user-images.githubusercontent.com/35104482/187878029-6fd2a819-957b-45e8-b3a2-888c00c984ca.png)
